### PR TITLE
[JENKINS-74032][JENKINS-73033][JENKINS-73034] Improve CSP compatibility

### PR DESCRIPTION
**Pull request by yaroslavafenkin**
* Extract inline script blocks and event handlers from `ComputerConfigHistoryAction/showDiffFiles.jelly`, `JobConfigHistoryProjectAction/showDiffFiles.jelly`, `JobConfigHistoryRootAction/showDiffFiles.jelly` to `show-diff-files-behaviour.js`
---

* https://issues.jenkins.io/browse/JENKINS-74032
* https://issues.jenkins.io/browse/JENKINS-74033
* https://issues.jenkins.io/browse/JENKINS-74034

### Testing done

[Before the change](https://www.youtube.com/watch?v=mVybddcPBus)

[After the change](https://www.youtube.com/watch?v=2qgq4-WP2fY)

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/resources/hudson/plugins/jobConfigHistory/JobConfigHistoryRootAction/resource.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+            document.getElementById("tbody_versionDiffsHidden").style.display = 'none';
+
+            button.value = hideText;
+        } else {
+            document.getElementById("tbody_versionDiffsShown").style.display = 'none';
+            document.getElementById("tbody_versionDiffsHidden").style.display = '';
+
+            button.value = showText;
+        }
+    }
```
Comment (yaroslavafenkin) [inline src/main/resources/hudson/plugins/jobConfigHistory/JobConfigHistoryRootAction/resource.js:19]:
```suggestion
    var showVersionDiffsJs = true;

    function toggleShowHideVersionDiffsJs(hideText, showText) {
        showVersionDiffsJs = !showVersionDiffsJs;
        if (showVersionDiffsJs === true) {
            document.getElementById("tbody_versionDiffsShown").style.display = '';
            document.getElementById("tbody_versionDiffsHidden").style.display = 'none';

            document.getElementById("showHideVersionDiffsJsButton").value = hideText;
        } else {
            document.getElementById("tbody_versionDiffsShown").style.display = 'none';
            document.getElementById("tbody_versionDiffsHidden").style.display = '';

            document.getElementById("showHideVersionDiffsJsButton").value = showText;
        }
    }
```

Diff with the old script.

File: src/main/resources/hudson/plugins/jobConfigHistory/JobConfigHistoryRootAction/resource.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    document.querySelector("#showHideVersionDiffsJsButton").addEventListener("click", (event) => {
+        const { hideText, showText } = event.target.dataset;
```
Comment (yaroslavafenkin) [inline src/main/resources/hudson/plugins/jobConfigHistory/JobConfigHistoryRootAction/resource.js:22]: This is well supported according to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#browser_compatibility.
I could switch to the old way of doing that though if that's too uncommon.

File: src/main/resources/hudson/plugins/jobConfigHistory/show-diff-files-behaviour.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        } else {
+            document.getElementById("tbody_versionDiffsShown").style.display = 'none';
+            document.getElementById("tbody_versionDiffsHidden").style.display = '';
+
+            button.value = showText;
+        }
+    }
+
+    document.querySelector("#showHideVersionDiffsJsButton").addEventListener("click", (event) => {
+        const { hideText, showText } = event.target.dataset;
```
Comment (yaroslavafenkin) [inline src/main/resources/hudson/plugins/jobConfigHistory/show-diff-files-behaviour.js:22]: This is well supported according to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#browser_compatibility.
I could switch to the old way of doing that though if that's too uncommon.

File: src/main/resources/hudson/plugins/jobConfigHistory/show-diff-files-behaviour.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+            document.getElementById("tbody_versionDiffsHidden").style.display = 'none';
+
+            button.value = hideText;
+        } else {
+            document.getElementById("tbody_versionDiffsShown").style.display = 'none';
+            document.getElementById("tbody_versionDiffsHidden").style.display = '';
+
+            button.value = showText;
+        }
+    }
```
Comment (yaroslavafenkin) [inline src/main/resources/hudson/plugins/jobConfigHistory/show-diff-files-behaviour.js:19]: Diff with the old script.
```suggestion
    var showVersionDiffsJs = true;

    function toggleShowHideVersionDiffsJs(hideText, showText) {
        showVersionDiffsJs = !showVersionDiffsJs;
        if (showVersionDiffsJs === true) {
            document.getElementById("tbody_versionDiffsShown").style.display = '';
            document.getElementById("tbody_versionDiffsHidden").style.display = 'none';

            document.getElementById("showHideVersionDiffsJsButton").value = hideText;
        } else {
            document.getElementById("tbody_versionDiffsShown").style.display = 'none';
            document.getElementById("tbody_versionDiffsHidden").style.display = '';

            document.getElementById("showHideVersionDiffsJsButton").value = showText;
        }
    }
```

**Comment by StefanSpieker** [comment]
Thanks for the PR, I was on vacation and will take a look over the weekend.

File: src/main/resources/hudson/plugins/jobConfigHistory/show-diff-files-behaviour.js (hunk 3 above)
Comment (basil) [reply src/main/resources/hudson/plugins/jobConfigHistory/show-diff-files-behaviour.js:22]: Spread syntax is indeed well-supported in real browsers and I have no problem with it. But please be careful with HtmlUnit-based tests, as HtmlUnit is not a real browser and the support matrix linked above does not apply to it. However, in this case I am pretty sure that HtmlUnit does support the spread syntax you are using here.

**Review by basil — APPROVED** [review]
Thanks for the PR!

**Pull request merged**